platform/platform-configuration-listener/src/main/java/org/codice/ddf/configuration/store/SystemConfigurationMigrator.java

```diff
@@ -0,0 +1,193 @@
+    private static final String TRUSTSTORE_SYSTEM_PROP = "javax.net.ssl.trustStore";
+
+    private static final String FILE_CONTAINING_CRL_LOCATION = "etc/ws-security/server/encryption.properties";
+
+    private static final String WS_SECURITY_DIR = "etc/ws-security";
+
+    private static final String SECURITY_DIRECTORY = "security";
+
+    private static final String PDP_POLICIES_DIR = "etc/pdp";
+
+    private static final String CRL_PROP_KEY = "org.apache.ws.security.crypto.merlin.x509crl.file";
+
+    private static final String SYSTEM_PROPERTIES = "etc/system.properties";
+
+    private static final String USERS_PROPERTIES = "etc/users.properties";
+
+    private final Path ddfHome;
+
+    public SystemConfigurationMigrator(Path ddfHome) {
+        this.ddfHome = ddfHome;
+    }
+
+    public void export(Path exportDirectory) throws MigrationException {
+        exportSecurity(exportDirectory);
+        exportSystemFiles(exportDirectory);
+    }
+
+    private void exportSecurity(Path exportDirectory) throws MigrationException {
+        exportSecurityDirectory(exportDirectory);
+        exportKeystores(exportDirectory);
+        exportWsSecurity(exportDirectory);
+        exportCrl(exportDirectory);
+        exportPdpPolicies(exportDirectory);
+    }
+
+    private void exportSystemFiles(Path exportDirectory) throws MigrationException {
+        exportSystemProperties(exportDirectory);
+        exportUsersProperties(exportDirectory);
+    }
+
+    private void exportKeystores(Path exportDirectory) throws MigrationException {
+        exportKeystore(exportDirectory);
+        exportTruststore(exportDirectory);
+    }
+
+    private void exportKeystore(Path exportDirectory) throws MigrationException {
+        String keystore = getProperty(KEYSTORE_SYSTEM_PROP);
+        if (StringUtils.isBlank(keystore)) {
+            throw new MigrationException(String.format(
+                    "Cannot find keystore. System property %s is not set.", KEYSTORE_SYSTEM_PROP));
+        }
+        Path source = ddfHome.resolve(keystore);
+        Path destination = constructDestination(source, exportDirectory);
+        copyFile(source, destination);
+    }
+
+    private void exportTruststore(Path exportDirectory) throws MigrationException {
+        String truststore = getProperty(TRUSTSTORE_SYSTEM_PROP);
+        if (StringUtils.isBlank(truststore)) {
+            throw new MigrationException(String.format(
+                    "Cannot find truststore. System property %s is not set.",
+                    TRUSTSTORE_SYSTEM_PROP));
+        }
+        Path source = ddfHome.resolve(truststore);
+        Path destination = constructDestination(source, exportDirectory);
+        copyFile(source, destination);
+    }
+
+    private void exportCrl(Path exportDirectory) throws MigrationException {
+        Path encryptionPropertiesFile = ddfHome.resolve(FILE_CONTAINING_CRL_LOCATION);
+        try (Stream<String> lines = getStream(encryptionPropertiesFile)) {
```

Comment:
Not sure if it would be simpler/clearer and help with the unit tests, but we could also do `new Properties.load(...).getProperty(CRL_PROP_KEY)` here instead of all the lambda expressions?